### PR TITLE
add visualize task to sbt

### PR DESCRIPTION
**Pull request by zartstrom**
.. from scala spree in CPH ..

We leverage the existing `graphSettings` function to create a task that writes dependency graphs to .dot-files.

See also this pdf created from `declared_dependencies.dot` output (created with `dot -Tpdf declared_dependencies.dot -o declared_deps.pdf`):
``
[declared_deps.pdf](https://github.com/sbt/sbt/files/1041913/declared_deps.pdf)

### CODE REVIEW

**Comment by lightbend-cla-validator** [comment]
Hi @zartstrom,

Thank you for your contribution! We really value the time you've taken to put this together.

Before we proceed with reviewing this pull request, please sign the Lightbend Contributors License Agreement:

[http://www.lightbend.com/contribute/cla](http://www.lightbend.com/contribute/cla)

File: sbt/src/sbt-test/project/visualize/test [hunk 1]
```diff
@@ -0,0 +1,6 @@
+# Run the visualize command and check for the output
+> visualize
+$ exists actual_dependencies.dot 
+$ exists declared_dependencies.dot 
+# $ copy-file actual_dependencies.dot /tmp
```
Comment (jvican) [inline sbt/src/sbt-test/project/visualize/test]: Can you remove these statements? We don't need to copy here :smile:.

**Review by jvican — APPROVED** [review]
LGTM, thank you Philipp for this contribution! I just requested a minor change.

**Comment by eed3si9n** [comment]
Perhaps this should exist first as a plugin to prove its usefulness before we merge this to sbt?
I would much rather have sbt-pgp or sbt-dependency-graph feature before generating a dot file.

**Comment by jvican** [comment]
> Perhaps this should exist first as a plugin to prove its usefulness before we merge this to sbt?
  
I think this feature was always meant to be provided by sbt, and Mark put all the infrastructure there in `Project.graphSettings`. The value of this plugin is that it allows to see the connections between all your task and settings. Sbt dependency graph and others already prove this feature is useful as is.

> I would much rather have sbt-pgp or sbt-dependency-graph feature before generating a dot file.
  
Sure, you can port them if you want too. I would like to have them merged into sbt.

**Comment by dwijnand** [comment]
I agree this should be spun out (and iterated on) externally to sbt first, in its own plugin. And if a useful form is found, we can reconsider folding it into sbt proper.

**Pull request closed without merging**